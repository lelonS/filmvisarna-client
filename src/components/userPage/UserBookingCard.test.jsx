import React, { useState as useStateMock } from 'react'
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import UserBookingCard from './UserBookingCard';

import { booking } from '../../../test-mocks/index.js';

// jest.mock('react', () => ({
//   ...jest.requireActual('react'),
//   useState: jest.fn(),
// }))





describe('UserBookingCard', () => {

  function expectBaseElementsToBeRendered() {
    let movie = booking.movie
    expect(screen.getByText(RegExp(movie.title, 'i'))).toBeInTheDocument();
    expect(screen.getByText('BokningsNr: ' + booking.bookingId)).toBeInTheDocument();
    expect(screen.getByAltText(`poster of the movie: ${booking.movie.title}`).getAttribute('src')).toBe('/img/' + movie.img_poster);
  }

  test('renders correctly when collapsed', () => {
    render(<UserBookingCard booking={booking} />);
    // Assert that the form elements are rendered correctly
    expectBaseElementsToBeRendered();
    // Expand buttons
    expect(screen.getByText('Mer info')).toBeInTheDocument();
    expect(screen.getByAltText('expand more icon')).toBeInTheDocument();
    expect(screen.queryByText('Mindre')).not.toBeInTheDocument();
    expect(screen.queryByAltText('expand less icon')).not.toBeInTheDocument();
  });

  test('renders correctly when expanded', () => {
    render(<UserBookingCard booking={booking} />);
    fireEvent.click(screen.getByText('Mer info'));

    // Assert that the form elements are rendered correctly
    expectBaseElementsToBeRendered();

    // Expand buttons
    expect(screen.getByText('Mindre')).toBeInTheDocument();
    expect(screen.getByAltText('expand less icon')).toBeInTheDocument();
    expect(screen.queryByText('Mer info')).not.toBeInTheDocument();
    expect(screen.queryByAltText('expand more icon')).not.toBeInTheDocument();

    // More info info
    expect(screen.getByRole('button', { name: 'Avboka' })).toBeInTheDocument();

    expect(screen.getByText(booking.screening.theaterName)).toBeInTheDocument();
    expect(screen.getByText('Rad:')).toBeInTheDocument();
    expect(screen.getByText('Plats:')).toBeInTheDocument();
    expect(screen.getByText('Pris:')).toBeInTheDocument();
    expect(screen.getByText('Status:')).toBeInTheDocument();
  });

  test('clicking Avboka button calls setCancelBooking and setToggle', () => {
    const setCancelBooking = jest.fn();
    const setToggle = jest.fn();
    render(<UserBookingCard booking={booking} setCancelBooking={setCancelBooking} setToggle={setToggle} />);
    fireEvent.click(screen.getByText('Mer info'));
    fireEvent.click(screen.getByRole('button', { name: 'Avboka' }));
    expect(setCancelBooking).toHaveBeenCalledTimes(1);
    expect(setCancelBooking).toHaveBeenCalledWith(booking);
    expect(setToggle).toHaveBeenCalledTimes(1);
  });

});