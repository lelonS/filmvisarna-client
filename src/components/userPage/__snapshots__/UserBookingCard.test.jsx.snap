// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBookingCard renders correctly when collapsed snapshot 1`] = `
<div
  className="flex justify-center rounded-lg text-white"
  style={{}}
>
  <div
    className="mx-auto w-full max-w-[17rem] xs:mx-0 xs:max-w-[24rem]"
  >
    <div
      className="flex flex-col justify-center xs:flex-row xs:items-end"
      style={{}}
    >
      <img
        alt="poster of the movie: The Creator"
        className="mx-auto w-[10rem] rounded-lg xs:mx-0"
        src="/img/theCreatorPoster.jpg"
      />
      <div
        className="mt-4 flex flex-col gap-1 xs:mb-0"
      >
        <p
          className="text-md ml-4 md:text-xl"
        >
          The Creator
        </p>
        <p
          className="ml-4 text-base md:text-lg"
        >
          SÃ¶ndag 24:e december
        </p>
        <p
          className="ml-4 text-base md:text-lg"
        >
          Klockan 
          21:00
        </p>
        <p
          className="ml-4 text-base md:text-lg"
        >
          BokningsNr: 
          9FA0HM
        </p>
        <div
          className="mx-auto mb-2 flex w-20 cursor-pointer whitespace-nowrap underline xs:mb-0"
          onClick={[Function]}
        >
          Mer info
          <img
            alt="expand more icon"
            className="w-8"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
